dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/icon/IconRequestParams.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2004-2024, University of Oslo
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.webapi.controller.icon;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.hisp.dhis.fieldfiltering.FieldFilterParser;
+import org.hisp.dhis.fieldfiltering.FieldPath;
+import org.hisp.dhis.icon.IconTypeFilter;
+
+/**
+ * @author Zubair Asghar
+ */
+@Data
+@NoArgsConstructor
+public class IconRequestParams {
+
+  static final String DEFAULT_FIELDS_PARAM =
+      "key,keywords,description,fileResourceUid,createdByUserUid,href";
```

Comment:
Depends whether default should be minimal or not;

The bare minimum to show would be key and fileResourceUid. This would be used by all consumers, since it's the minimum required to link an icon reference (key) to a file (fileResourceUid).

However, keywords and descriptions are useful by adding more context, so if you're browsing the api, this makes sense to include; But for our internal apps, it's primarily relevant for maintenance app (Unless we want ALT text for images). And if we starting thinking we add anything relevant for maintenance, I would say this is a good default.

This I feel would be more useful to get feedback from the frontend teams instead (@vgarciabnz @Birkbjo @JoakimSM @janhenrikoverland ) But if there is no preference, I'd be fine with what is in this list :)

Reply:
It looks fine to me. Anyway, the Android app always explicitly includes the list of fields in the request to avoid downloading unused properties.

Reply:
> Depends whether default should be minimal or not;

I think the default should be a sensible minimum 😁 like we don't return nested entities in tracker export by default to avoid fetching everything just because someone is browsing the API 😁 

Any app should use the approach @vgarciabnz in Android takes. Always specify the fields you need.